packages/std/src/addresses.rs

```diff
@@ -42,49 +53,111 @@ impl Addr {
     /// let address = Addr::unchecked("foobar");
     /// assert_eq!(address, "foobar");
     /// ```
-    pub fn unchecked<T: Into<String>>(input: T) -> Addr {
-        Addr(input.into())
+    pub const fn unchecked(input: &str) -> Addr {
+        let input_bytes = input.as_bytes();
+        let mut data = [0u8; ADDR_MAX_LENGTH];
+        let mut i = 0;
+        while i < input_bytes.len() {
```

Comment:
```suggestion
        while i < input_bytes.len() && i < ADDR_MAX_LENGTH {
```

Reply:
This is in a fact a little more convoluted:
```
        let length = input_bytes.len();
        let length = if length < ADDR_MAX_LENGTH {
            length
        } else {
            ADDR_MAX_LENGTH
        };
        while i < length {
            data[i] = input_bytes[i];
            i += 1;
        }

        Addr(AddrInner {
            data,
            length,
        })
```
As we'd want both, avoid out of bounds panics, and assign consistent addresses.

Reply:
Or, given that `unchecked()` is thought to be used in test code, and for already validated addresses, are you OK with out of bounds panicking it?

Reply:
The panic was intended. I'd rather crash hard than silently change data. Unfortunately there is no panic for `const fn` stabilized (https://rust-lang.github.io/rfcs/2345-const-panic.html).

Reply:
Makes sense; particularly for `unchecked`.